### PR TITLE
A few follow up changes to LookupTypeKey change

### PR DESCRIPTION
**Pull request by VSadov**
As a followup to the LookupTypeKey PR (https://github.com/dotnet/runtime/pull/61346) I looked at possibility of reducing nested types hash collisions in `EEClassHashTable`  by mixing hashes of enclosers into the hash. 

That was implemented in https://github.com/dotnet/runtime/pull/61652, but I was not convinced the results justify added complexity. 
I do not think we should take the whole https://github.com/dotnet/runtime/pull/61652.

So here I have a smaller PR with just few unambiguous improvements that I came up with while working on #61652 . I do not think we want the rest of the change.

### CODE REVIEW

File: src/coreclr/vm/clsload.cpp [hunk 1]
```diff
@@ -1238,18 +1243,18 @@ BOOL ClassLoader::FindClassModuleThrowing(
...
     {
+        // Take the lock. To make sure the table is not being built by another thread.
         AvailableClasses_LockHolder lh(this);
 
-        // Try again with the lock.  This will protect against another thread reallocating
-        // the hash table underneath us
-        GetClassValue(nhTable, pName, &Data, &pTable, pLookInThisModuleOnly, &foundEntry, loadFlag, needsToBuildHashtable);
-        pBucket = foundEntry.GetClassHashBasedEntryValue();
-
+        if (!needsToBuildHashtable || (m_cUnhashedModules == 0))
```
Comment (VSadov) [inline src/coreclr/vm/clsload.cpp:1249]: We do not need to do lookup 3 times if tables were populated by another thread when we did a lookup. 
Just one retry is sufficient in all cases.

File: src/coreclr/vm/clsload.cpp [hunk 2]
```diff
@@ -1536,19 +1541,19 @@ ClassLoader::LoadTypeHandleThrowing(
...
-            typeHnd.IsRestored() &&
-            foundEntry.GetEntryType() == HashedTypeEntry::EntryType::IsHashedClassEntry &&
-            (foundEntry.GetClassHashBasedEntryValue() != NULL) &&
-            (foundEntry.GetClassHashBasedEntryValue()->GetData() != typeHnd.AsPtr()))
-        {
-            foundEntry.GetClassHashBasedEntryValue()->SetData(typeHnd.AsPtr());
-        }
-#endif // !DACCESS_COMPILE
+    // Replace AvailableClasses Module entry with found TypeHandle
+    if (!typeHnd.IsNull() &&
```
Comment (VSadov) [inline src/coreclr/vm/clsload.cpp:1548]: This is an optimization where we can replace a token with a type handle. That would save a lookup in another table.
There is some additional infrastructure in the table to support this and this is very intentional, but I think someone misplaced `}` so this code never ran. 

All successful lookups break from the loop, so the code taking the note of the found handle must be outside.

File: src/coreclr/vm/clsload.cpp [hunk 3]
```diff
@@ -719,6 +716,9 @@ void ClassLoader::GetClassValue(NameHandleTable nhTable,
             }
             _ASSERTE(pTable);
 
+            mdToken             mdEncloser;
+            BOOL isNested = IsNested(pName, &mdEncloser);
```
Comment (VSadov) [inline src/coreclr/vm/clsload.cpp:720]: `IsNested` is a bit involved. In some cases like R2R, we may not need `IsNested`, so it makes sense to move the call down to where we are sure we need it.

**Comment by VSadov** [comment]
@jkotas - this is just couple changes from https://github.com/dotnet/runtime/pull/61652, that I think are worth taking on their own. Please take a look.

**Review by jkotas — APPROVED** [review]
Thanks

**Comment by VSadov** [comment]
Thanks!1

**Pull request merged**